### PR TITLE
Dependency upgrades

### PR DESCRIPTION
**Pull request by Alf-Melmac**
## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [x] Other: Gradle and Dependency updates

## Description

Jar sizes before and after updates (in KB) _(Build date: 03.11.2021)_
| Jar | Before | After | Factor |
| --- | --- | --- | --- |
| DEV | 1440 | 1440 | 1 |
| DEV-javadoc | 2983 | 2983 | 1 |
| DEV-sources | 1235 | 1235 | 1 |
| DEV-withDependencies | 10369 | 12530 | 1.2 | 
| DEV-withDependencies-min | 4624 | 6409 | 1.4 |
| DEV-withDependencies-no-opus | 7066 | 9227 | 1.3 |

Dependency Upgrades
| Dependency with changelog | Old version | New version | Comment |
| --- | --- | --- | --- |
| [com.github.johnrengelman.shadow](https://github.com/johnrengelman/shadow/releases) | 5.1.0 | 7.1.2 | Required Gradle 7.0+ |
| [org.jetbrains:annotations](https://github.com/JetBrains/java-annotations/releases) | 16.0.1 | 23.0.0 | |
| [org.slf4j:slf4j-api](http://www.slf4j.org/news.html) | 1.7.25 | 1.7.36 | |
| [com.squareup.okhttp3:okhttp](https://square.github.io/okhttp/changelog/#version-491) | 3.13.0 | 4.9.3 | |
| [org.apache.commons:commons-collections4](https://commons.apache.org/proper/commons-collections/release_4_4.html) | 4.1 | 4.4 | |
| [org.junit.jupiter:junit-jupiter](https://junit.org/junit5/docs/5.8.2/release-notes/) | 5.4.0 | 5.8.2 | |
| [Gradle](https://docs.gradle.org/7.4/userguide/upgrading_version_7.html#changes_7.4) | 7.3.3 | 7.4.2 | Update from 6.8.3 -> 7.3.3 already done in #1976 |
| com.fasterxml.jackson.core:jackson-core | - | 2.13.2 | |
| com.fasterxml.jackson.core:jackson-databind | 2.10.1 | 2.13.2.2 | |

### CODE REVIEW

**Comment by Alf-Melmac** [comment]
This needs some more testing. So far everything looks good, there have been no breaking changes, Gradle excluded, that affect us.

**Comment by thejakwe** [comment]
nitpick: this should also bump the versions at the bottom of the readme

**Comment by DV8FromTheWorld** [comment]
Out of curiosity, do we know if it is the upgrade to Okhttp that is spiking our jar sizes?

**Comment by Chew** [comment]
Definitely a factor...

[Okhttp 3.13.0](https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp/3.13.0) is [404 KB](https://repo1.maven.org/maven2/com/squareup/okhttp3/okhttp/3.13.0).
[Okhttp 4.9.1](https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp/4.9.1) is [779 KB](https://repo1.maven.org/maven2/com/squareup/okhttp3/okhttp/4.9.1/).

Also concerning:
[JNA 4.4.0](https://mvnrepository.com/artifact/net.java.dev.jna/jna/4.4.0) is [1.0 MB](https://repo1.maven.org/maven2/net/java/dev/jna/jna/4.4.0)
[JNA 5.9.0](https://mvnrepository.com/artifact/net.java.dev.jna/jna/5.9.0) is [1.6 MB]

**Comment by HeyBanditoz** [comment]
Doesn't OkHttp 4.x pull Kotlin as a dependency also?

**Comment by duncte123** [comment]
> > 
> > 
> > Doesn't OkHttp 4.x pull Kotlin as a dependency also?
> 
> According to https://square.github.io/okhttp/upgrading_to_okhttp_4/#gradle , there should be no difference to 3.13.

That only talks about minimum requirements. And yes, it does pull Kotlin.

**Comment by BrooklynDewolf** [comment]
~~#￼. ￼en nu~~~~￼~~~~~
#lnb_____ik _e~~~~~~~~[]()[[]()]()
  E~~e~~~~~~eeree~~~~~~~~o
Ek￼

File: build.gradle.kts [hunk 1]
```diff
@@ -119,6 +119,7 @@ dependencies {
 
     //General Utility
     implementation("net.sf.trove4j:trove4j:3.0.3")
+    // Match the dependency version of lavaplayer
     implementation("com.fasterxml.jackson.core:jackson-databind:2.10.1")
```
Comment (MinnDevelopment) [inline build.gradle.kts:123]: We should update this to get rid of the warning on mvnrepository. Also, can you add a dependency on `jackson-core` as well?

Comment (Alf-Melmac) [reply build.gradle.kts]: To maintain [compatibility with Lavaplayer](https://github.com/DV8FromTheWorld/JDA/pull/1819#discussion_r708700892), I updated the version to the same minor version, but to a version without CVE.
Matching jackson-core version added

**Review by MinnDevelopment — APPROVED** [review]

File: build.gradle.kts [hunk 2]
```diff
@@ -122,15 +122,17 @@ dependencies {
 
     //General Utility
     implementation("net.sf.trove4j:trove4j:3.0.3")
-    implementation("com.fasterxml.jackson.core:jackson-databind:2.10.1")
+    // Match the minor version of lavaplayers jackson dependency
```
Comment (DV8FromTheWorld) [inline build.gradle.kts:125]: This comment isn't right anymore.

File: build.gradle.kts (hunk 1 above)
Comment (Alf-Melmac) [reply build.gradle.kts]: We [agreed](https://discord.com/channels/125227483518861312/869965829024915466/965660685700841533) to kill this CVE and upgrade to 2.13.2(.2). Lavaplayer will follow soon

**Pull request merged**